cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/OIDCUtil.java

```diff
@@ -325,17 +326,22 @@ public static String buildIDToken(RelyParty rp, boolean isUserInfo) throws Excep
         HttpPrincipal useridPrincipal = subject.getPrincipals(HttpPrincipal.class).iterator().next();
         String email = OIDCUtil.getEmail(useridPrincipal);
         Calendar calendar = Calendar.getInstance();
-        
+        String clientID = rp.getClientID();
+
         if (rp.isSignDocuments() || !isUserInfo) {
             JwtBuilder builder = Jwts.builder();
-            builder.claim("sub", numericPrincipal.getName());
-            builder.claim("iss", getClaimIssuer());
-            if (rp.getClientID() != null) {
-                builder.claim("aud", rp.getClientID());
-            }
-            builder.claim("iat", calendar.getTime());
+
+            builder.setAudience(getClaimIssuer())   // aud
+                .setIssuedAt(calendar.getTime())    // iat
+                .setIssuer(clientID)                // iss
+                .setSubject(numericPrincipal.getName()) // sub
+                .setHeaderParam("typ", "JWT")
+                .setHeaderParam("alg", "RSA256");
```

Comment:
I'm still surprised that the 'audience' is us and not the rely party, and that the 'issuer' is rely party and not us.  The spec seems to say it's this way too:  https://openid.net/specs/openid-connect-core-1_0.html#IDToken

Reply:
I'll switch that back - somewhere I'd found one example that appeared to read differently, but I can't find it again. Ultimately as long as the value put in & tested for are the same, that's all that seems to matter.